lib/compat/wordpress-6.2/widgets.php

```diff
@@ -0,0 +1,32 @@
+<?php
+/**
+ * Core Widget APIs for WP 6.2.
+ *
+ * @package gutenberg
+ */
+
+if ( ! function_exists( '_wp_block_theme_stub_sidebars' ) ) {
+	/**
+	 * Register the previous theme's sidebars for the block themes.
+	 *
+	 * @return void
+	 */
+	function _wp_block_theme_stub_sidebars() {
+		global $wp_registered_sidebars;
+
+		if ( ! wp_is_block_theme() ) {
+			return;
+		}
+
+		$legacy_sidebars = get_theme_mod( 'wp_legacy_sidebars' );
+		if ( empty( $legacy_sidebars ) ) {
+			return;
+		}
+
+		// Don't use `register_sidebar` since it will enable the `widgets` support for a theme.
+		foreach ( $legacy_sidebars as $sidebar ) {
+			$wp_registered_sidebars[ $sidebar['id'] ] = $sidebar;
```

Comment:
> It's kind of unexpected that hitting /wp/v2/sidebars on a block theme would return sidebars though, no?

* The endpoint already returns the "Inactive Widgets," I think marking and returning "old" sidebars for block themes as `inactive` makes sense here.
* If we change the `POST` request behavior, it should be the same for any inactive sidebar.

Reply:
Cool looks like that works.

```sh
$ http -a admin:password GET http://wp-git-build.test/wp-json/wp/v2/sidebars | jq '.[] | {id, status, widgets}'
{
  "id": "wp_inactive_widgets",
  "status": "inactive",
  "widgets": []
}
{
  "id": "sidebar-1",
  "status": "inactive",
  "widgets": [
    "block-2",
    "block-3",
    "block-4",
    "block-5",
    "block-6",
    "block-7",
    "meta-3",
    "color-picker-2",
    "sow-button-2"
  ]
}
```

I think that makes sense. Would love a second opinion from @TimothyBJacobs.

It would be nice to have some amount of unit test for the REST API logic and the theme switching logic. Anything to do with widgets tends to break if there are no tests because it gets forgotten by contributors. 

> If we change the POST request behavior, it should be the same for any inactive sidebar.

What do you mean by this? Updating the REST API so that modifying _any_ inactive sidebar fails?

Reply:
> I know it's a pain but anything to do with widgets tends to break if there are no tests because it gets forgotten by contributors.

This is one of the reasons I'm trying to keep changes to the Widgets and Sidebars API to the minimum 😅 I'm going to add integration/unit tests in PHPUnit.

> > If we change the POST request behavior, it should be the same for any inactive sidebar.
>
> What do you mean by this? Updating the REST API so that modifying any inactive sidebar fails?

Maybe we should leave the sidebar's endpoint `POST` behavior as it is. Then, if users with the right capabilities want to modify sidebar widgets with "handcrafted" API requests, we should let them do it.

Reply:
> This is one of the reasons I'm trying to keep changes to the Widgets and Sidebars API to the minimum 😅

You'll be okay! I'm here to help 😀 (Also there are tests that should catch most regressions.)

> Maybe we should leave the sidebar's endpoint POST behavior as it is. Then, if users with the right capabilities want to modify sidebar widgets with "handcrafted" API requests, we should let them do it.

Sure but if the API consumer makes any changes to these inactive sidebars then those changes will be clobbered next time the API is accessed because line 28 here will overwrite whatever is in `$wp_registered_sidebars`. This is why I'm thinking just disable mutating inactive sidebars altogether.

Reply:
The widgets API sure is fun 🙂 I agree, this is a workable solution given out constraints.

Reply:
Thanks for the feedback, @TimothyBJacobs.